src/main/java/com/minecolonies/api/entity/citizen/AbstractEntityCitizen.java

```diff
@@ -233,6 +235,18 @@ public boolean isNoAi()
         return false;
     }
 
+    @Override
+    @Nullable
+    protected PlayerTeam getAssignedTeam()
+    {
+        final ICitizenColonyHandler citizenColonyHandler = getCitizenColonyHandler();
+        if (citizenColonyHandler == null || citizenColonyHandler.getColony() == null)
```

Comment:
No method for the client side?

Reply:
This logic happens to work identically for the client, so this works just fine.

Reply:
Did we recently rework the citizenColonyHandler? I was sure it only worked on the server

Reply:
the registerWithColony call wasn't made for the client side, e.g. getCitizenmanager.registerCivilian